x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -74,6 +70,33 @@ async function syncDataForAllLayers(getState, dispatch, dataFilters) {
   await Promise.all(syncs);
 }
 
+export function trackCurrentLayerState(layerId) {
+  return {
+    type: TRACK_CURRENT_LAYER_STATE,
+    layerId: layerId
+  };
+}
+
+export function rollbackToTrackedLayerStateForSelectedLayer() {
```

Comment:
This needs to dispatch `syncDataForLayer` for the selected layer

Reply:
good catch.